src/vs/workbench/contrib/preferences/browser/settingsWidgets.ts

```diff
@@ -525,46 +548,85 @@ export class ListSettingWidget extends AbstractListSettingWidget<IListDataItem>
 		const valueElement = DOM.append(rowElement, $('.setting-list-value'));
 		const siblingElement = DOM.append(rowElement, $('.setting-list-sibling'));
 
-		valueElement.textContent = item.value;
+		valueElement.textContent = item.value.data.toString();
 		siblingElement.textContent = item.sibling ? `when: ${item.sibling}` : null;
 
 		return rowElement;
 	}
 
 	protected renderEdit(item: IListDataItem, idx: number): HTMLElement {
 		const rowElement = $('.setting-list-edit-row');
+		let valueInput: InputBox | SelectBox;
+		let currentDisplayValue: string;
+		let currentEnumOptions: IObjectEnumOption[] | undefined;
 
-		const updatedItem = () => ({
-			value: valueInput.value,
-			sibling: siblingInput?.value
-		});
+		if (this.isItemNew(item) && this.keyValueSuggester) {
+			const enumData = this.keyValueSuggester(this.model.items.map(({ value: { data } }) => data));
+			item = {
+				...item,
+				value: {
+					type: 'enum',
+					data: item.value.data,
+					options: enumData ? enumData.options : []
+				}
+			};
+		}
 
+		switch (item.value.type) {
+			case 'string':
+				valueInput = this.renderInputBox(item.value, rowElement);
+				break;
+			case 'enum':
+				valueInput = this.renderDropdown(item.value, rowElement);
+				currentEnumOptions = item.value.options;
+				if (item.value.options.length) {
+					currentDisplayValue = currentEnumOptions[0].value;
```

Comment:
Not sure it's always correct to pick option 0 here? Can another index be the default selection?

Reply:
Not sure what was going on here. I changed it to `item.value.data`, since that is the current value to display for that item.